### DISCUSSION TITLE
Substrate 通证 JS 方储存方法

### DISCUSSION THREAD

**Discussion by jimmychu0807** [ParityAsia/substrate-newsletter#5] [Substrate/Polkadot 技术问题]
我想咨询一下， bn.js 最大支持2^53的数，如果1 Unit的token小数点后15位，那么1 unit =10^15, 1000 unit = 10^18 > 2^53，这种情况下javascript api 应该怎么把这么大的balance （比如1000 unit ) 发送过去呢?

我看前端上面有kunit, munit, 不过不知道直接用javascript传的时候怎么弄?

 \- 小马哥

**Comment by jimmychu0807** [comment]
从 [这里](https://github.com/polkadot-js/api/blob/master/packages/types/package.json#L24) 可看出我们确是用 bn.js。

不过 bn.js 看文档，他们声称可支持无限大的数目，而不是 2^53, 看 [这里](https://www.npmjs.com/package/bn.js#number-size)。当你要转到 `toNumber()` 时则受限于  53 bits, 所以处理帐号余额，尽量别转换到 `toNumber()` 去。 我相信 [polkadot-js API types](https://github.com/polkadot-js/api/tree/master/packages/types) 是实现了它自己的方法来转换到显示方便的格式，而没有用 `toNumber()`. 具体要在代码里找。

**Reply by chaosma** [reply]
我目前用的是 javascript自带的算术，然后转换成BigInt，没有报错了。我查了一下api 支持BN, BigNumber, 好几种格式
